### ISSUE TITLE
HttpHelper readFully fails when talking with last version of Oxalis.

### ISSUE THREAD

**Issue by mstennie** [phax/as2-lib#64]
Hello Philip,

We encountered an error using your code in Babelway while talking with Oxalis 4.0.2 and above.

We replaced HttpHelper call to read the chuncked payload by new ChunkedInputStream(is) provided by commons-httpclient 3.1 library. 

Mathieu Stennier.
Dev @ Babelway.

**Comment by phax** [comment]
Hi Mathieu, thanks for your input.
You may not have noticed, but there will be a version 4.2.0 which supports large files, which implies, that HTTPClient will be used.
So what was the issue? That you MUST used chunked encoding? Or is it an Oxalis bug?
If this is too sensitive, we can also talk on Slack.
// Philip

**Comment by mstennie** [comment]
We encountered an EOFException in the readFully method. Most probably the calculation of the chunkLength was not working well.

I don't think it is an Oxalis bug because everything worked well by using the ChunkedInputStream of the commons library. 

We integrated your changes first for support of large files but still encountered the error so we decided to remove the call to readPayload and just extract the bytes by doing:

new ChunkedInputStream(httpRequest.getInputStream());

Hope this helps,

Mathieu Stennier.
